kaldb/src/main/proto/kaldb_configs.proto

```diff
@@ -164,3 +168,34 @@ message ManagerConfig {
 message RecoveryConfig {
   ServerConfig server_config = 1;
 }
+
+// Config for the preprocessor node.
+message PreprocessorConfig {
+  // Configuration for the kafka stream processor
+  message KafkaStreamConfig {
+    string bootstrap_servers = 1;
+    // An identifier for the stream processing application. Must be unique within the Kafka cluster
+    string application_id = 2;
+    // The number of threads to execute stream processing
+    int32 num_stream_threads = 3;
+    // This will allow parallel processing up to the amount of upstream partitions. You cannot have
+    // more threads than you have upstreams due to how the work is partitioned. E
+    string processing_guarantee = 4;
+  }
+
+  ServerConfig server_config = 1;
+  KafkaStreamConfig kafka_stream_config = 2;
+
+  // Upstream topics to consume from
+  repeated string upstream_topics = 3;
+
+  // Downstream topic to write to
+  string downstream_topic = 4;
```

Comment:
As discussed this is currently the case. We'll follow up this PR with an optional way to set a separate downstream using the `forEach` instead of a `to`.